src/stdlib_experimental_quadrature_simps.fypp

```diff
@@ -0,0 +1,277 @@
+#:include "common.fypp"
+
+submodule (stdlib_experimental_quadrature) stdlib_experimental_quadrature_simps
+
+    implicit none
+
+    ! internal use only
+    interface simps38
+    #:for k1, t1 in REAL_KINDS_TYPES
+        module procedure simps38_dx_${k1}$
+        module procedure simps38_x_${k1}$
+    #:endfor
+    end interface simps38
+
+    ! internal use only
+    interface simps38_weights
+    #:for k1, t1 in REAL_KINDS_TYPES
+        module procedure simps38_weights_${k1}$
+    #:endfor
+    end interface simps38_weights
+
+contains
+
+#:for k1, t1 in REAL_KINDS_TYPES
+
+    pure recursive module function simps_dx_${k1}$(y, dx, even) result(integral)
+        ${t1}$, dimension(:), intent(in) :: y
+        ${t1}$, intent(in) :: dx
+        integer, intent(in), optional :: even
+        ${t1}$ :: integral
+
+        integer :: n
+
+        n = size(y)
+
+        select case (n)
+        case (0:1)
+            integral = 0.0_${k1}$
+        case (2)
+            integral = 0.5_${k1}$*dx*(y(1) + y(2))
+        case (3)
+            integral = dx/3.0_${k1}$*(y(1) + 4*y(2) + y(3))
+        case (4)
+            integral = simps38(y, dx)
+        ! case (5) not needed; handled by default
```

Comment:
What should happen if I provide a array `y` of size > 6? Currently it will go to the `default case` and all the entries (i.e. more than 5) will be used to compute `integral`.
Instead, should a warning or an error be mentioned? Or is it the intended behaviour?
I usually use `default case` to report such "errors".

Reply:
The default case handles everything that's not a special case. Here, special cases arise for small arrays, which need special-cased quadrature rules or control flow to do the right thing. It is the nature of these types of quadrature rules to use all the array entries given.

Reply:
OK. Thank you for the explanations.